job/runner.go

```diff
@@ -85,6 +99,45 @@ func (j *JobRunner) Run(cache JobCache) (*JobStat, Metadata, error) {
 	return j.currentStat, j.meta, nil
 }
 
+// LocalRun executes the Job's local shell command
+func (j *JobRunner) LocalRun() error {
+	return j.runCmd()
+}
+
+// RemoteRun sends a http request, and checks if the response is valid in time,
+func (j *JobRunner) RemoteRun() error {
+	// Calculate a response timeout
+	timeout := j.responseTimeout()
+
+	httpClient := http.Client{
+		Timeout: timeout,
+	}
+
+	// Normalize the method passed by the user
+	method := strings.ToUpper(j.job.RemoteProperties.Method)
+	bodyBuffer := bytes.NewBufferString(j.job.RemoteProperties.Body)
+	req, err := http.NewRequest(method, j.job.RemoteProperties.Url, bodyBuffer)
+	if err != nil {
+		return err
+	}
+
+	// Set default or user's passed headers
+	j.setHeaders(req)
+
+	// Do the request
+	res, err := httpClient.Do(req)
+	if err != nil {
+		return err
+	}
+
+	// Check if we got any of the status codes the user asked for
+	if j.checkExpected(res.StatusCode) {
+		return nil
+	} else {
+		return errors.New(res.Status)
+	}
+}
+
```

Comment:
On line 145, CircleCI says

    args, err := shParser.Parse(j.job.Command)

is racing with some other line. Investigating now...